File: emagro/app_emagro/lib/models/usuario.dart
class Usuario {
  final int id;
  final String nombre;
  final String usuario;
  final String rol;
  final String estado;
  final String? fechaCreacion;
  final String? fechaActualizacion;

  Usuario({
    required this.id,
    required this.nombre,
    required this.usuario,
    required this.rol,
    required this.estado,
    this.fechaCreacion,
    this.fechaActualizacion,
  });

  // Crear Usuario desde JSON
  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: int.parse(json['id'].toString()),
      nombre: json['nombre'] ?? '',
      usuario: json['usuario'] ?? '',
      rol: json['rol'] ?? '',
      estado: json['estado'] ?? '',
      fechaCreacion: json['fecha_creacion'],
      fechaActualizacion: json['fecha_actualizacion'],
    );
  }

  // Convertir Usuario a JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'usuario': usuario,
      'rol': rol,
      'estado': estado,
      'fecha_creacion': fechaCreacion,
      'fecha_actualizacion': fechaActualizacion,
    };
  }

  // Verificar si es administrador
  bool get isAdmin => rol == 'admin';

  // Verificar si está activo
  bool get isActivo => estado == 'activo';

  @override
  String toString() {
    return 'Usuario{id: $id, nombre: $nombre, usuario: $usuario, rol: $rol, estado: $estado}';
  }
}

File: BasquetSanarate/Backend/src/Entities/Usuario.php
<?php
namespace App\Entities;

class Usuario
{
    public function __construct(
        public ?int $id = null,
        public string $usuario = '',
        public string $nombre = '',
        public string $rol = 'admin',
        public bool $activo = true
    ) {
    }

    public static function fromRow(array $row): self
    {
        return new self(
            (int) $row['id'],
            $row['usuario'],
            $row['nombre'],
            $row['rol'] ?? 'admin',
            (bool) ($row['activo'] ?? 1)
        );
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'usuario' => $this->usuario,
            'nombre' => $this->nombre,
            'rol' => $this->rol,
            'activo' => $this->activo,
        ];
    }
}
